sdk/servicebus/service-bus/test/browserified/topicFilters.spec.ts

```diff
@@ -0,0 +1,589 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import chai from "chai";
+const should = chai.should();
+import chaiAsPromised from "chai-as-promised";
+import * as dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+import {
+  ServiceBusClient,
+  SubscriptionClient,
+  ServiceBusMessage,
+  TopicClient,
+  SendableMessageInfo,
+  CorrelationFilter,
+  ReceiveMode
+} from "../../src";
+import {
+  getSenderReceiverClients,
+  TestClientType,
+  purge,
+  checkWithTimeout
+} from "./testUtils";
+
+// We need to remove rules before adding one because otherwise the existing default rule will let in all messages.
+async function removeAllRules(client: SubscriptionClient): Promise<void> {
+  const rules = await client.getRules();
+  for (let i = 0; i < rules.length; i++) {
+    const rule = rules[i];
+    await client.removeRule(rule.name);
+  }
+}
+
+async function testPeekMsgsLength(
+  client: SubscriptionClient,
+  expectedPeekLength: number
+): Promise<void> {
+  const peekedMsgs = await client.peek(expectedPeekLength + 1);
+  should.equal(
+    peekedMsgs.length,
+    expectedPeekLength,
+    "Unexpected number of msgs found when peeking"
+  );
+}
+
+let ns: ServiceBusClient;
+let subscriptionClient: SubscriptionClient;
+let topicClient: TopicClient;
+
+async function beforeEachTest(receiverType: TestClientType): Promise<void> {
+  // The tests in this file expect the env variables to contain the connection string and
+  // the names of empty queue/topic/subscription that are to be tested
+
+  // @ts-ignore
+  if (!window.__env__["SERVICEBUS_CONNECTION_STRING"]) {
+    throw new Error(
+      "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+    );
+  }
+
+  // @ts-ignore
+  ns = ServiceBusClient.createFromConnectionString(window.__env__["SERVICEBUS_CONNECTION_STRING"]);
+
+  const clients = await getSenderReceiverClients(
+    ns,
+    TestClientType.TopicFilterTestTopic,
+    receiverType
+  );
+  topicClient = clients.senderClient as TopicClient;
+  subscriptionClient = clients.receiverClient as SubscriptionClient;
+
+  await purge(subscriptionClient);
+  const peekedSubscriptionMsg = await subscriptionClient.peek();
+  if (peekedSubscriptionMsg.length) {
+    chai.assert.fail("Please use an empty Subscription for integration testing");
+  }
+  if (receiverType === TestClientType.TopicFilterTestSubscription) {
+    await removeAllRules(subscriptionClient);
+  }
+}
+
+async function afterEachTest(clearRules: boolean = true): Promise<void> {
+  if (clearRules) {
+    await removeAllRules(subscriptionClient);
+    await subscriptionClient.addRule("DefaultFilter", true);
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "Unexpected number of rules");
+    should.equal(rules[0].name, "DefaultFilter", "RuleName is different than expected");
+  }
+  await ns.close();
+}
+
+const data = [
```

Comment:
As mentioned in https://github.com/Azure/azure-sdk-for-js/issues/2163#issuecomment-491172859, skip tests that involve sending of data in this file

Reply:
Unclear what this is referring to, will sync up offline

Reply:
The linked comment had the details as to why we dont need to add tests that send and receive messages in this file. Repeating here for more clarity

Any tests that are sending messages and then testing the filter by looking at the received messages in this test file are not required as that is a service feature. Since the sdk does nothing special in this area, one round of tests for node are enough. Replicating them for browser is not needed.

Reply:
For topic filter tests, in addition to the tests we discussed in person about excluding, I have also excluded the tests that were testing negative scenarios going by the intent of testing if errors are thrown.

In general, every test sends some data for testing. The intent of filtering tests that purely rely on sending of Service Bus messages is now clear.
This seems generally time consuming as it would require going through each test case and not sure if it would be worth doing so for all the tests. 

In general, our SDK should test code in SDK and not purely the service at all. 
And if it does touch any SDK code, then I don't see why we shouldn't test the same for browserified SDKs (even more so, because we have introductions such as isNode and checks of the likes that may impact code paths).

Reply:
Resolving this comment thread as the requested changes are made.  We can use https://github.com/Azure/azure-sdk-for-js/pull/2775#discussion_r284510429 for concerns around the sdk vs service, node vs browser.